src/satosa/backends/apple.py

```diff
@@ -192,24 +201,49 @@ def response_endpoint(self, context, *args):
         :return:
         """
         backend_state = context.state[self.name]
-        authn_resp = self.client.parse_response(AuthorizationResponse, info=context.request, sformat="dict")
+
+        # Apple sends some user information only via POST in the first request
+        if "user" in context.request:
+            userinfo = json.load(context.request["user"])
+            userinfo["name"] = " ".join(
```

Comment:
I don't think `name` should be constructed here. This is the job of a micro-service. Here we are only collecting the userinfo and claims with their protocol-specific meaning, as sent by the issuer.

Reply:
So you would do this instead?
```py
userinfo = json.load(context.request["user"])
for namePart in ["firstName", "middleName", "lastName"]:
  userinfo[namePart] = userinfo.get(namePart, "")
```